frappe/desk/search.py

```diff
@@ -219,29 +213,32 @@ def search_widget(
 				else (cint(ignore_user_permissions) and has_permission(doctype, ptype=ptype))
 			)
 
-			if doctype in translated_doctypes:
-				page_length = None
-
 			values = frappe.get_list(
 				doctype,
 				filters=filters,
 				fields=formatted_fields,
 				or_filters=or_filters,
 				limit_start=start,
-				limit_page_length=page_length,
+				limit_page_length=None if meta.translated_doctype else page_length,
 				order_by=order_by,
 				ignore_permissions=ignore_permissions,
 				reference_doctype=reference_doctype,
 				as_list=not as_dict,
 				strict=False,
 			)
 
-			if doctype in translated_doctypes:
+			if meta.translated_doctype:
 				# Filtering the values array so that query is included in very element
 				values = (
 					v
 					for v in values
 					if re.search(f"{re.escape(txt)}.*", _(v.name if as_dict else v[0]), re.IGNORECASE)
+					or (
+						# If we show the title, also search in translated titles
+						re.search(f"{re.escape(txt)}.*", _(v.label if as_dict else v[1]), re.IGNORECASE)
+						if show_title_field
+						else False
+					)
```

Comment:
- Should we search all the fields ?? In link fields, for en, we search through all the search fields for a query; why not do the same here?

Reply:
Yeah, if it doesn't hurt the performance too much.